src/main/java/com/fabriccommunity/spookytime/mixin/ThrownPotionEntityMixin.java

```diff
@@ -1,26 +1,65 @@
 package com.fabriccommunity.spookytime.mixin;
 
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.FlyingItemEntity;
 import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
 import net.minecraft.entity.mob.BlazeEntity;
 import net.minecraft.entity.mob.EndermanEntity;
 import net.minecraft.entity.mob.WitchEntity;
+import net.minecraft.entity.thrown.ThrownEntity;
 import net.minecraft.entity.thrown.ThrownPotionEntity;
+import net.minecraft.util.math.Box;
+import net.minecraft.world.World;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
 /**
- * Causes witches to be damaged by splash potions of water
+ * Rewrites the water damage functionality of thrown potions to allow witches to be damaged by splash potions of water
  *
  * @author Will Toll
  */
 
 @Mixin(ThrownPotionEntity.class)
-public class ThrownPotionEntityMixin {
-	@Inject(method = "doesWaterHurt", at = @At("RETURN"))
-	private static boolean doesWaterHurt(LivingEntity livingEntity_1, CallbackInfoReturnable<Boolean> info) {
-		return livingEntity_1 instanceof EndermanEntity || livingEntity_1 instanceof BlazeEntity || livingEntity_1 instanceof WitchEntity;
+public abstract class ThrownPotionEntityMixin
+	extends ThrownEntity
+	implements FlyingItemEntity {
+
+	private List<Class<? extends LivingEntity>> hurtByWater = new ArrayList<Class<? extends LivingEntity>>();
+
+	public ThrownPotionEntityMixin(EntityType<? extends ThrownPotionEntity> type, World world) {
+		super(type, world);
+		registerHurtByWater(EndermanEntity.class);
+		registerHurtByWater(BlazeEntity.class);
+		registerHurtByWater(WitchEntity.class);
+	}
+
+	private void damageEntitiesHurtByWater() {
+		this.world.getEntities(LivingEntity.class, this.getBoundingBox().expand(4.0D, 2.0D, 4.0D)).forEach((entity) -> {
+			if (this.squaredDistanceTo(entity) < 16.0D && isHurtByWater(entity)) {
+				entity.damage(DamageSource.magic(entity, this.getOwner()), 1.0F);
+			}
+		});
+	}
+
+	public void registerHurtByWater(Class<? extends LivingEntity> entityclass) {
+		this.hurtByWater.add(entityclass);
+	}
+
+	private boolean isHurtByWater(LivingEntity entity) {
+		for (Class<? extends LivingEntity> entityclass : this.hurtByWater) {
+			if (entity.getClass().isInstance(entityclass)) {
+				return true;
+			}
+		}
+		return true;
 	}
```

Comment:
Do any of these overwrite or override a vanilla method?

Reply:
I don’t think this is even a mixin anymore, I see no objections, redirections, shadowing, modifying variables, etc.

Reply:
In fact I don’t think the methods in this class are even used anywhere else, it sort of seems pointless overall.

Reply:
Kind of, so damageEntitiesHurtByWater overwrites a method in ThrownPotionEntity, but the ThrownPotionEntityMixin class by Java's standards only extends ThrownEntity so it would throw an error if the @Override annotation was added. In terms of functionality however it does entirely rewrite the way that the original method in ThrownPotionEntity.class functions, but it does it in a safe-ish way because the only time anyone else would want to overwrite that method is if they wanted to add a new entity to be affected by it, but that is handled already by giving the ThrownPotionEntity.class a public method to register new entities via the mixin. Hopefully that made sense.

Reply:
Then use `@Shadow` to get an instance of the list in ThrownPotionEntity and add your entries to it. No need to reinvent the wheel.

Reply:
ThrownPotionEntity doesn't utilize a list by default, it's all new functionality. Minecraft by default hard codes it as a static method which is why all of those methods had to be added.

Reply:
Then where is this list even being used. You don’t mixin anywhere to use it.

Reply:
And overriding methods is very bad practice, this could easily be done via injection.

Reply:
The list is a new variable that will get mixed into ThrownPotionEntity, from there the new public method registerHurtByWater allows classes that extend LivingEntity to be registered to be affected by splash potions of water. When a splash potions of water is thrown ThrownEntityPotion already calls damageEntitiesHurtByWater to damage them. That method has been completely rewritten to trigger the new isHurtByWater method to check if the targeted entity is registered in the ThrownPotionEntity's list of entities that are hurt by water.

Reply:
My thing is I don’t see a scenario where we in this mod would want to add functionality to other mobs to be hurt by water potion. Additionally, this breaks usage other mods from changing what mobs are hurt by water. This approach I am not very fond of, there was nothing wrong with your previous approach other than returning values in your injection the wrong way.

Reply:
Please don't overwrite (mixin's `@Overwrite` is assumed if no other annotation is present) target class methods in mixins. That causes crashing incompatibilities when somebody tries to inject to that method. Fabric mods tend to use cancellable injections at `HEAD` that always cancel after the logic, but if you can avoid overwriting, you should.

Reply:
The issue with using injection is that the original method first asks the world to only send it entities that meet the existing static variable. So if I were to only use injection that call to the world would still be using the old static functionality and not the new dynamic registry functionality.

Reply:
As I said, you can cancel right after your logic to soft-overwrite the method.

Reply:
Simply put, I will not approve this PR unless we do not Overwrite any methods and only use ways of adding this functionality that do not overwrite vanilla behavior or break functionality with other mods.

Reply:
That is simply unacceptable.